scipy/integrate/_ivp/ivp.py

```diff
@@ -26,28 +27,49 @@ class OdeResult(OptimizeResult):
 
 
 def prepare_events(events):
-    """Standardize event functions and extract is_terminal and direction."""
+    """Standardize event functions and extract the 'terminal', 'max_events',
+    and 'direction' event function attributes."""
     if callable(events):
         events = (events,)
 
-    if events is not None:
-        is_terminal = np.empty(len(events), dtype=bool)
-        direction = np.empty(len(events))
-        for i, event in enumerate(events):
+    max_events = np.empty(len(events))
+    direction = np.empty(len(events))
+    for i, event in enumerate(events):
+        try:
+            terminal = event.terminal
+            terminate_after = event.max_events
+            raise RuntimeError("Only one of 'terminal' and 'max_events' "
+                               "attributes can be supplied for a given event.")
+        except AttributeError:
+            pass
+
+        try:
+            terminal = event.terminal
+            warnings.warn(
+                "The 'terminal' event function attribute is deprecated and "
+                "will be removed in SciPy release 1.8.0. Please use the new "
+                "'max_events' attribute, which can be set to 1 to replicate "
+                "the behaviour of the 'terminal' attribute.",
+                DeprecationWarning, stacklevel=3)
```

Comment:
Perhaps I'm missing something, but `1.8.x` series is pretty soon (December-ish)--was there a discussion about using such a fast feature removal cycle if you're just deprecating it now? Usually two releases before you can remove I think?

Reply:
I have to say I put `1.8.0` because I did not know what else to put. Should it say something like "will be removed in a future SciPy release" instead?

Reply:
The next feature release will be 1.9, so as of today please put 1.11. (two feature releases later)